src/app/extensions/punchout/pages/punchout/punchout-page.guard.ts

```diff
@@ -48,6 +50,7 @@ export class PunchoutPageGuard implements CanActivate {
 
     // initiate the punchout user login with the access-token (cXML) or the given credentials (OCI)
     if (route.queryParamMap.has('access-token')) {
+      this.identityProvider.getInstance().triggerLogout();
       this.accountFacade.loginUserWithToken(route.queryParamMap.get('access-token'));
     } else {
       this.accountFacade.loginUser({
```

Comment:
Is the logout to clean up previous user session information only needed for cXML punchout login (with access token) or wouldn't this be needed for the OCI punchout with username and password as well?

Reply:
You're right, I modified the behavior to include OCI